book/build.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+from pathlib import Path
+import urllib.parse
+
+
+SCRIPT_DIR = Path(__file__).parent
+
+
+def generate_example(filename, title, strip_assets):
+    path_in = SCRIPT_DIR.parent / "examples" / f"{filename}.nbt"
+    path_out = SCRIPT_DIR / "src" / f"example-{filename}.md"
+
+    code = []
+    with open(path_in, "r") as fin:
+        for line in fin:
+            if not strip_assets or "assert_eq" not in line:
+                code.append(line)
+
+    url = f"https://numbat.dev/?q={urllib.parse.quote_plus(''.join(code))}"
+
+    with open(path_out, "w") as fout:
+        fout.write("<!-- This file is autogenerated! Do not modify it -->\n")
+        fout.write("\n")
+        fout.write(f'# {title} <a href="{url}"><i class="fa fa-play"></i></a>\n')
```

Comment:
Nice idea with the icon! Layout-wise, I think I would prefer something a bit more conservative like:
```suggestion
        fout.write(f"# {title}\n")
        fout.write(f'<a href="{url}"><i class="fa fa-play"></i> Run this example</a>\n')
```

Reply:
Sounds good to me!